### PR TITLE
Adding pet type object to pets list and pet details

### PR DESCRIPTION
**Pull request by autodidaddict**
Signed-off-by: Kevin Hoffman <autodidaddict@users.noreply.github.com>

### CODE REVIEW

**Review by ks2211 — APPROVED** [review]
lgtm

**Comment by autodidaddict** [comment]
@ks2211 once a code owner approves we can merge.

File: petclinic/actors/clinicapi/src/lib.rs
Comment (brooksmtownsend) [inline petclinic/actors/clinicapi/src/lib.rs:278]: I'm just imagining a robo-pet when I see `AugmentedPet`

**Review by brooksmtownsend — APPROVED** [review]

File: petclinic/actors/clinicapi/src/lib.rs
Comment (autodidaddict) [reply petclinic/actors/clinicapi/src/lib.rs:278]: :robot: :dog:

Comment (autodidaddict) [reply petclinic/actors/clinicapi/src/lib.rs:278]: That's a _very different_ kind of pet clinic then :grin:

**Pull request merged**
